docs/syntax_and_semantics/blocks_and_procs.md

```diff
@@ -132,19 +132,19 @@ end
 # Output: 3
 ```
 
-It's an error specifying more block arguments than those yielded:
+It's an error specifying more block parameters than the arguments yielded:
 
 ```crystal
 def twice
   yield
   yield
 end
 
-twice do |i| # Error: too many block arguments
+twice do |i| # Error: too many block parameters
```

Comment:
This change is not correct until the compiler error message is updated.

Reply:
A _lot_ other error messages are also outdated, so I simply assume that they do not correspond to exact error messages and that we should revisit them later.

Reply:
Well the idea is to represent the actual error message.

Reply:
But I suppose we can apply this change in expectation that the compiler error message is going to change as well as suggested in crystal-lang/crystal#10374.

Reply:
The error messages are now covered by crystal-lang/crystal#10378 (although this particular one might also be affected by the changes in crystal-lang/crystal#10251 so I updated that as well).